src/local-history/local-history-tree-provider.ts

```diff
@@ -0,0 +1,72 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as moment from 'moment';
+// eslint-disable-next-line no-unused-vars
+import { LocalHistoryManager } from './local-history-manager';
+
+export class Revision extends vscode.TreeItem {
+    uri: string;
+
+    constructor(label: string, uri: string, command: vscode.Command) {
+        super(label);
+        this.label = label;
+        this.uri = uri;
+        this.command = command;
+    }
+
+    get tooltip(): string {
+        return `Open '${path.basename(this.uri)}'`;
+    }
+
+    get description(): string {
+        return `Last modified ${moment(this.label!.replace(/[-: ]/g, ''), 'YYYYMMDDhhmmss').fromNow()}`;
+    }
+}
+
+export class LocalHistoryTreeProvider implements vscode.TreeDataProvider<Revision> {
+    private _onDidChangeTreeData: vscode.EventEmitter<Revision | undefined> = new vscode.EventEmitter<Revision | undefined>();
+    readonly onDidChangeTreeData: vscode.Event<Revision | undefined> = this._onDidChangeTreeData.event;
+    provider: LocalHistoryManager;
```

Comment:
[minor] should it not be `manager`?

Reply:
@vince-fugnitto I also updated all the `provider` in extension.ts to `manager`.